File: src/main/java/net/ccbluex/liquidbounce/utils/render/shader/shaders/GlowShader.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.utils.render.shader.shaders;

import net.ccbluex.liquidbounce.utils.render.shader.FramebufferShader;
import org.lwjgl.opengl.GL20;

public final class GlowShader extends FramebufferShader {

    public static final GlowShader GLOW_SHADER = new GlowShader();

    public GlowShader() {
        super("glow.frag");
    }

    @Override
    public void setupUniforms() {
        setupUniform("texture");
        setupUniform("texelSize");
        setupUniform("color");
        setupUniform("fade");
        setupUniform("radius");
        setupUniform("targetAlpha");
    }

    @Override
    public void updateUniforms() {
        GL20.glUniform1i(getUniform("texture"), 0);
        GL20.glUniform2f(getUniform("texelSize"), 1F / mc.displayWidth * renderScale, 1F / mc.displayHeight * renderScale);
        GL20.glUniform3f(getUniform("color"), red, green, blue);
        GL20.glUniform1f(getUniform("fade"), fade);
        GL20.glUniform1i(getUniform("radius"), radius);
        GL20.glUniform1f(getUniform("targetAlpha"), targetAlpha);
    }
}

File: src/main/java/net/ccbluex/liquidbounce/utils/render/shader/shaders/GlowShader.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.utils.render.shader.shaders

import net.ccbluex.liquidbounce.utils.render.shader.FramebufferShader
import org.lwjgl.opengl.GL20.*

object GlowShader : FramebufferShader("glow.frag") {
    
    override fun setupUniforms() {
        setupUniform("texture")
        setupUniform("texelSize")
        setupUniform("color")
        setupUniform("fade")
        setupUniform("radius")
        setupUniform("targetAlpha")
    }

    override fun updateUniforms() {
        glUniform1i(getUniform("texture"), 0)
        glUniform2f(getUniform("texelSize"),
            1f / mc.displayWidth * renderScale,
            1f / mc.displayHeight * renderScale
        )
        glUniform3f(getUniform("color"), red, green, blue)
        glUniform1f(getUniform("fade"), fade.toFloat())
        glUniform1i(getUniform("radius"), radius)
        glUniform1f(getUniform("targetAlpha"), targetAlpha)
    }


}
